pkg/providers/vsphere/defaults.go

```diff
@@ -101,59 +101,74 @@ func (d *Defaulter) setupDefaultTemplate(ctx context.Context, spec *Spec, machin
 	return nil
 }
 
-func (d *Defaulter) setCloneModeDefaults(ctx context.Context, machineConfig *anywherev1.VSphereMachineConfig, datacenter string) error {
-	diskSize, err := d.govc.GetVMDiskSizeInGB(ctx, machineConfig.Spec.Template, datacenter)
+func max(a, b int) int {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+func (d *Defaulter) setCloneModeAndDiskSizeDefaults(ctx context.Context, machineConfig *anywherev1.VSphereMachineConfig, datacenter string) error {
+	templateDiskSize, err := d.govc.GetVMDiskSizeInGB(ctx, machineConfig.Spec.Template, datacenter)
 	if err != nil {
 		return fmt.Errorf("getting disk size for template %s: %v", machineConfig.Spec.Template, err)
 	}
 
+	minDiskSize := max(minDiskGib, templateDiskSize)
+
+	if machineConfig.Spec.DiskGiB < minDiskSize {
+		logger.Info("Warning: VSphereMachineConfig DiskGiB cannot be less than %s. Defaulting to %s.", minDiskSize, minDiskSize)
```

Comment:
We need to change this to an fmt.Sprintf, I will follow up with a PR with this change.

Reply:
https://github.com/aws/eks-anywhere/pull/4883